Functional.Common.Extensions/ResultPartition.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace Functional;
+
+public readonly struct ResultPartition<TSuccess, TFailure>
+	where TSuccess : notnull
+	where TFailure : notnull
+{
+	internal ResultPartition(IEnumerable<TSuccess> successCollection, IEnumerable<TFailure> failureCollection)
+	{
+		SuccessCollection = successCollection ?? throw new ArgumentNullException(nameof(successCollection));
+		FailureCollection = failureCollection ?? throw new ArgumentNullException(nameof(failureCollection));
+	}
+
+	public IEnumerable<TSuccess> SuccessCollection { get; }
```

Comment:
For consistency with the Union partition classes, would be you fine with naming these `Successes` and `Failures`. Personally I think I prefer how that would read in a expression as well.

Reply:
I can make that change, yes.  I default to a `Collection` suffix because when skim-reading it stands out more than `s`.